src/main/java/com/auth0/utils/tokens/RS256SignatureVerifier.java

```diff
@@ -0,0 +1,49 @@
+package com.auth0.utils.tokens;
+
+import com.auth0.exception.IdTokenValidationException;
+import com.auth0.exception.PublicKeyProviderException;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.interfaces.JWTVerifier;
+import com.auth0.jwt.interfaces.RSAKeyProvider;
+
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+
+/**
+ * An implementation of {@code SignatureVerifier} for tokens signed with the RS256 asymmetric signing algorithm.
+ */
+class RS256SignatureVerifier extends SignatureVerifier {
+
+    RS256SignatureVerifier(PublicKeyProvider publicKeyProvider) {
+        super(createJWTVerifier(publicKeyProvider));
+    }
+
+    private static JWTVerifier createJWTVerifier(final PublicKeyProvider publicKeyProvider) {
+        Algorithm alg = Algorithm.RSA256(new RSAKeyProvider() {
+            @Override
+            public RSAPublicKey getPublicKeyById(String keyId) {
+                try {
+                    return publicKeyProvider.getPublicKeyById(keyId);
+                } catch (PublicKeyProviderException pke) {
+                    throw new IdTokenValidationException("Error retrieving public key", pke);
```

Comment:
checking with our table of messages, this should be:
```suggestion
                    throw new IdTokenValidationException(String.format("Could not find a public key for Key ID (kid) \"%s\"", keyId), pke);
```

I just typed that here 🤞